crates/services/consensus_module/poa/src/pre_confirmation_signature_service/broadcast.rs

```diff
@@ -1,19 +1,38 @@
-use super::*;
-use std::future::Future;
+use super::Result;
+use crate::pre_confirmation_signature_service::{
+    parent_signature::ParentSignature,
+    signing_key::SigningKey,
+};
+use core::future::Future;
+use fuel_core_types::services::p2p::DelegatePreConfirmationKey;
+
+#[allow(type_alias_bounds)]
+pub type PublicKey<S>
+where
+    S: Broadcast,
+= <S::DelegateKey as SigningKey>::PublicKey;
+
+#[allow(type_alias_bounds)]
+pub type Signature<S>
+where
+    S: Broadcast,
+= <S::DelegateKey as SigningKey>::Signature;
 
 /// Broadcasts the delegate key as well as the pre-confirmed transactions to the network
 pub trait Broadcast: Send {
-    type PreConfirmations: Send + Clone + Serialize;
-    type ParentSignature;
+    type ParentKey: ParentSignature;
```

Comment:
I think this should still be named `ParentSignature`. We never take ownership of the key. Maybe `ParentKeySigner`.

Reply:
Meh. This is fine.